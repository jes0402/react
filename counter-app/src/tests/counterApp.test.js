import React from 'react';
import CounterApp from '../counterApp'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';


describe ("pruebas en <CounterApp />", () => {
    let wrapper = shallow(<CounterApp />);;

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    })

    test ("debe de mostrar <CounterApp /> correctamente", () => {

        const wrapper = shallow(<CounterApp />);

        expect(wrapper ).toMatchSnapshot();

    })
    test ("debe de mostrar el valor por defecto de 100", () => {
      
    const wrapper = shallow( <CounterApp valor={ 100 }/> );
    const counterText = wrapper.find("h2").text().trim();
     expect( counterText ).toBe('100');

})
    test ("debe de incrementar con el boton +1", () => {
        wrapper.find("button").at(0).simulate('click');
        const counterText = wrapper.find("h2").text().trim();
        expect( counterText ).toBe('11')
    })
    test ("debe de colocar el valor por defecto con el boton Reset", () => {
        const wrapper = shallow( <CounterApp valor={ 100 }/> );
        wrapper.find("button").at(0).simulate('click');
        wrapper.find("button").at(1).simulate('click');
        const counterText = wrapper.find("h2").text().trim();
        expect( counterText ).toBe('100')
    })
    test ("debe decrementar con el boton -1", () => {
        wrapper.find("button").at(2).simulate('click');
        const counterText = wrapper.find("h2").text().trim();
        expect( counterText ).toBe("9")
    })
});