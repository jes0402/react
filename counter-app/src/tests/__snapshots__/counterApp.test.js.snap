// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <CounterApp /> debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</Fragment>
`;
